applications/GeoMechanicsApplication/tests/cpp_tests/custom_utilities/test_neighbouring_element_finder.cpp

```diff
@@ -0,0 +1,268 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#include "custom_utilities/neighbouring_element_finder.hpp"
+#include "test_setup_utilities/element_setup_utilities.h"
+#include "test_setup_utilities/model_setup_utilities.h"
+#include "tests/cpp_tests/geo_mechanics_fast_suite.h"
+
+namespace
+{
+using namespace Kratos;
+
+void CreateNumberOfNewNodes(ModelPart& rModelPart, std::size_t NumberOfNodes)
+{
+    for (std::size_t i = 0; i < NumberOfNodes; ++i) {
+        rModelPart.CreateNewNode(i + 1, 0.0, 0.0, 0.0);
+    }
+}
+
+PointerVector<Node> GetNodesFromIds(ModelPart& rModelPart, const std::vector<std::size_t>& rNodeIds)
+{
+    PointerVector<Node> result(rNodeIds.size());
+    std::ranges::transform(rNodeIds, result.ptr_begin(),
+                           [&rModelPart](auto Id) { return rModelPart.pGetNode(Id); });
+    return result;
+}
+
+} // namespace
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(NeighbouringElementFinder_ReturnsEmptyListWhenNoNeighbouringElements,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    Model model;
+    auto& r_model_part_for_neighbouring_elements = model.CreateModelPart("empty");
+
+    NeighbouringElementFinder finder;
+
+    auto& r_model_part_for_entities_for_finding =
+        ModelSetupUtilities::CreateModelPartWithASingle2D3NElement(model);
+
+    std::map<std::size_t, std::unique_ptr<BoundaryGenerator>> boundary_generators;
```

Comment:
For clarity, we may want to reuse the alias that you've defined for this map:
```suggestion
    NeighbouringElementFinder::BoundaryGeneratorByLocalDim boundary_generators;
```
This suggestion also applies to the other unit tests.

Reply:
Good suggestion, done